File: app/src/main/java/com/byd/dashcast/infrastructure/MirrorDaemonReusePolicy.java
package com.byd.dashcast.infrastructure;

/** Decides whether an existing MirrorDaemon is safe to keep across app lifecycle changes. */
final class MirrorDaemonReusePolicy {
    private MirrorDaemonReusePolicy() {}

    static int singleProcessPid(String psOutput) {
        if (psOutput == null || psOutput.trim().isEmpty()) return -1;
        String[] lines = psOutput.trim().split("\\r?\\n");
        if (lines.length != 1) return -1;
        String[] fields = lines[0].trim().split("\\s+");
        if (fields.length < 2) return -1;
        try {
            return Integer.parseInt(fields[1]);
        } catch (NumberFormatException ignored) {
            return -1;
        }
    }

    static boolean shouldReuse(boolean binderAlive, int daemonPid,
                               String versionMarker, int appBuild) {
        if (!binderAlive || daemonPid <= 0 || versionMarker == null) return false;
        String[] fields = versionMarker.trim().split(":", -1);
        if (fields.length != 2) return false;
        try {
            int markerPid = Integer.parseInt(fields[0]);
            int daemonBuild = Integer.parseInt(fields[1]);
            return markerPid == daemonPid && daemonBuild == appBuild;
        } catch (NumberFormatException ignored) {
            return false;
        }
    }
}

File: app/src/main/java/com/byd/dashcast/infrastructure/MirrorDaemonReusePolicy.kt
package com.byd.dashcast.infrastructure

/** Decides whether an existing MirrorDaemon is safe to keep across app lifecycle changes. */
object MirrorDaemonReusePolicy {
    @JvmStatic
    fun singleProcessPid(psOutput: String?): Int {
        if (psOutput == null || psOutput.trim().isEmpty()) return -1
        val lines = psOutput.trim().split(Regex("\\r?\\n"))
        if (lines.size != 1) return -1
        val fields = lines[0].trim().split(Regex("\\s+"))
        if (fields.size < 2) return -1
        return try {
            fields[1].toInt()
        } catch (ignored: NumberFormatException) {
            -1
        }
    }

    @JvmStatic
    fun shouldReuse(binderAlive: Boolean, daemonPid: Int, versionMarker: String?, appBuild: Int): Boolean {
        if (!binderAlive || daemonPid <= 0 || versionMarker == null) return false
        val fields = versionMarker.trim().split(":")
        if (fields.size != 2) return false
        return try {
            val markerPid = fields[0].toInt()
            val daemonBuild = fields[1].toInt()
            markerPid == daemonPid && daemonBuild == appBuild
        } catch (ignored: NumberFormatException) {
            false
        }
    }
}
